### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by mcbloch**
also fix a classname lookup on a 404.

## Usage in code

No annotation means you don't need to be authenticated
`@PermitAll` means all authenticated users can access the resource
`@DenyAll` means nobody can access it
`@Auth Optional<User> userOpt` as path parameter means authentication is not required and gives you a user object, even for anonymous users. Useful to to remove authentication for 1 method when auth is required for the rest of the class

For more see the docs https://www.dropwizard.io/en/latest/manual/auth.html

## Using the api

Authenticate using basic auth specifying a username:password. The password is a common one at the moment. When it's valid, your specified username can be used for further authorization.
Password at the moment is **'secret'**.
All Create, Update and Delete methods now require authentication.

### Examples

Valid authentication:
`http -a username:secret localhost:8080/hello-world/auth`

Invalid authentication:
`http -a username:invalid localhost:8080/hello-world/auth`

Using curl:
`curl -u username:secret localhost:8080/hello-world/auth`

### CODE REVIEW

File: src/main/java/telraam/App.java [hunk 1]
```diff
@@ -86,11 +88,17 @@ public void run(AppConfiguration configuration, Environment environment)
...
 
         // Add basic authentication
+        Map<String, String> validCredentials = config.getApplicationCredentials().stream()
+                .collect(Collectors.toMap(
+                        AppConfiguration.ApplicationCredentialFactory::getUsername,
+                        AppConfiguration.ApplicationCredentialFactory::getPassword));
         environment.jersey().register(new AuthDynamicFeature(
                 new BasicCredentialAuthFilter.Builder<User>()
                         .setAuthenticator(credentials -> {
                             // If the password is 'secret' then create a user with the specified username
```
Comment (Robbe7730) [inline src/main/java/telraam/App.java:107]:
```suggestion

```

**Review by Robbe7730 — CHANGES_REQUESTED** [review]
Minor nitpick, looks good otherwise :slightly_smiling_face:

**Comment by abeforgit** [comment]
note this is breaking for all currently existing clients

**Comment by abeforgit** [comment]
recommend to merge _after_ test-day

**Comment by redfast00** [comment]
We've decided to not do authentication

**Pull request closed without merging**